api/base/generic_bulk_views.py

```diff
@@ -0,0 +1,110 @@
+
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework_bulk import generics as bulk_generics
+from rest_framework.exceptions import PermissionDenied, ValidationError
+
+from framework.auth.core import Auth
+
+from website.project.model import Q
+from api.base.settings import BULK_SETTINGS
+from api.base.exceptions import Conflict, JSONAPIException
+from api.base.utils import is_bulk_request
+
+
+class ListBulkCreateJSONAPIView(bulk_generics.ListBulkCreateAPIView):
+    """
+    Custom ListBulkCreateAPIView that properly formats bulk create responses
+    in accordance with the JSON API spec
+    """
+
+    # overrides ListBulkCreateAPIView
+    def create(self, request, *args, **kwargs):
+        """
+        Correctly formats both bulk and single POST response
+        """
+        if is_bulk_request(request):
+            if not request.data:
+                raise ValidationError('Request must contain array of resource identifier objects.')
+
+        response = super(ListBulkCreateJSONAPIView, self).create(request, *args, **kwargs)
+        if 'data' in response.data:
+            return response
+        return Response({'data': response.data}, status=status.HTTP_201_CREATED)
+
+    # overrides ListBulkCreateAPIView
+    def get_serializer(self, *args, **kwargs):
+        """
+        Adds many=True to serializer if bulk operation.
+        """
+
+        if is_bulk_request(self.request):
+            kwargs['many'] = True
+
+        return super(ListBulkCreateJSONAPIView, self).get_serializer(*args, **kwargs)
+
+
+class BulkUpdateJSONAPIView(bulk_generics.BulkUpdateAPIView):
+    """
+    Custom BulkUpdateAPIView that properly formats bulk update responses in accordance with
+    the JSON API spec
+    """
+
+    # overrides BulkUpdateAPIView
+    def bulk_update(self, request, *args, **kwargs):
+        """
+        Correctly formats bulk PUT/PATCH response
+        """
+        if not request.data:
+            raise ValidationError('Request must contain array of resource identifier objects.')
+
+        response = super(BulkUpdateJSONAPIView, self).bulk_update(request, *args, **kwargs)
+        return Response({'data': response.data}, status=status.HTTP_200_OK)
```

Comment:
See previous comment. It might be better to overwrite the existing response's data attribute than to re-instantiate a response.